partitura/score.py

```diff
@@ -252,6 +252,45 @@ def measure_map(self):
 
         return lambda x: measures[np.searchsorted(measures[:, 1], x, side="right"), :]
 
+    @property
+    def measure_number_map(self):
+        """A function mapping timeline times to the measure number of
+        the measure they are contained in. The function can take
+        scalar values or lists/arrays of values.
+
+        Returns
+        -------
+        function
+            The mapping function
+
+        """
+        measures = np.array([(m.start.t, m.end.t, m.number) for m in self.iter_all(Measure)])
+
+        # correct for anacrusis
+        divs_per_beat = self.inv_beat_map(
+            1 + self.beat_map(0)
+        )  # find the divs per beat in the first measure
+        if (
+            measures[0][1] - measures[0][0]
+            < self.time_signature_map(0)[0] * divs_per_beat
+        ):
+            measures[0][0] = (
+                measures[0][1] - self.time_signature_map(0)[0] * divs_per_beat
+            )
+
+        if len(measures) == 0:  # no measures in the piece
+            # default only one measure spanning the entire timeline
+            LOGGER.warning("No measures found, assuming only one measure")
+            if self.first_point is None:
+                t0, tN = 0, 0
+            else:
+                t0 = self.first_point.t
+                tN = self.last_point.t
+
+            measures = np.array([(t0, tN, 1)])
+
+        return lambda x: measures[np.searchsorted(measures[:, 1], x, side="right"), 2]
```

Comment:
Maybe using interp1d instead of `lambda` and `searchsorted` could be faster? I don't know if this is the case, though, so you can disregard this comment ;)

Reply:
This follows the structure of ```measure_map```.

Reply:
I think we should reconsider this lambda everywhere then. First, it is kind of counter-intuitive to call `part.func()` empty and then add another parenthesis for the arguments. Second, adding a lambda makes the new lambda function anonymous so makes it also hard for debugging. A solution could be to use `functools.partial` or just define all part functions to receive divs argument. Is there any particular reason I am not aware of for not doing this?

Reply:
I'm not familiar with functools.partial and interp1d. If you think it will be faster we can change that (also in measure_map). We just need to make sure that this can work both with one and multiple input values.

Reply:
Both maps have property decorators so the double parentheses problem isn't there. I'm not familiar with partial functions either, but it's pretty straightforward to use interp1d and will work with single and array inputs, so I can change that.